app/code/Magento/ConfigurableProduct/Model/Product/Type/VariationMatrix.php

```diff
@@ -40,7 +40,9 @@ public function getVariations($usedProductAttributes)
             for ($attributeIndex = $attributesCount; $attributeIndex--;) {
                 $currentAttribute = $variationalAttributes[$attributeIndex];
                 $currentVariationValue = $currentVariation[$attributeIndex];
-                $filledVariation[$currentAttribute['id']] = $currentAttribute['values'][$currentVariationValue];
+                    if(isset($currentAttribute['id']) && $currentAttribute['id']) {
```

Comment:
You may simplify the expression to `!empty($currentAttribute['id'])` according to https://www.php.net/manual/en/function.empty.php

Reply:
Sure, done